### PR TITLE
refactor: optimize perf-counter

### PR DESCRIPTION
**Pull request by Shuo-Jia**
The perf-counter has some problem:
* Existed many redundant counter, such as `*.cps-15min`,`latency.p50` etc,  this pr delete these.
* No existed suitable tag: user usually don't set the tag config or we don‘t get it though it is set, this pr will add **first ip of meta list** and table name as default tag which also join to the user tag if it is set. **add `meta=ip` tag is because sometimes different meta_server has same table name, so that if you search `table=name` , we will not distinguish it**. The follow is an tag example:

> user tag string: cluster=xiaomi
> the final tag : cluster=xiaomi,meta=ip,table=name

* `enable_perf_counter` default value is set `true`

### CODE REVIEW

**Comment by acelyc111** [comment]
> this pr will add **first ip of meta list** and table name as default tag which also join to the user tag if it is set.

cluster meta list is mutable, how to deal with the case when meta server changed?

**Comment by Shuo-Jia** [comment]
> > this pr will add **first ip of meta list** and table name as default tag which also join to the user tag if it is set.
> 
> cluster meta list is mutable, how to deal with the case when meta server changed?

Yeah, I have considered it. Actually, only `table tag` is included in different clusters, meta ip is apply. now, since we meta list is not host name, I haven't find a way the same as sever client to tag one cluster.

File: src/main/java/com/xiaomi/infra/pegasus/metrics/MetricsPool.java [hunk 1]
```diff
@@ -98,27 +82,28 @@ public String metricsToJson() throws JSONException {
...
     builder.append("]");
 
     return builder.toString();
   }
 
+  private String getTableTag(String counterName) {
+    String[] result = counterName.split("@");
+    if (result.length >= 2) {
+      String tableName = result[1].split("\\.")[0];
+      return defaultTag.equals("") ? "table=" + tableName : defaultTag + ",table=" + tableName;
```
Comment (neverchanje) [inline src/main/java/com/xiaomi/infra/pegasus/metrics/MetricsPool.java:102]:
```suggestion
      if(theMetric.tags.empty() || !theMetric.tags.contains("table=")) {
        // add tag "table=<tableName>"
        return theMetric.tags + ",table=" + tableName;
      }
```

**Comment by Shuo-Jia** [comment]
New commit has delete meta ip

File: src/main/java/com/xiaomi/infra/pegasus/metrics/MetricsPool.java [hunk 2]
```diff
@@ -98,27 +82,30 @@ public String metricsToJson() throws JSONException {
...
     return builder.toString();
   }
 
+  private String getTableTag(String counterName) {
+    if (defaultTag.contains("table=")) {
+      return defaultTag;
+    }
+    String[] result = counterName.split("@");
+    if (result.length >= 2) {
+      return defaultTag.equals("") ? "table=" + result[1] : defaultTag + ",table=" + result[1];
```
Comment (levy5307) [inline src/main/java/com/xiaomi/infra/pegasus/metrics/MetricsPool.java:104]: `defaultTag.equals("") ? "table=" + result[1] : (defaultTag + ",table=" + result[1]);`

Comment (levy5307) [inline src/main/java/com/xiaomi/infra/pegasus/metrics/MetricsPool.java:104]: `return defaultTag + ",table=" + result[1];`

Comment (levy5307) [inline src/main/java/com/xiaomi/infra/pegasus/metrics/MetricsPool.java:104]: return defaultTag.equals("") ? "table=" + result[1] : (defaultTag + ",table=" + result[1]);

**Comment by Shuo-Jia** [comment]
I will re-test it manually before merged

File: src/main/java/com/xiaomi/infra/pegasus/rpc/async/ClientRequestRound.java [hunk 3]
```diff
@@ -34,12 +34,15 @@ public ClientRequestRound(
...
-        cb,
-        enableCounter,
-        System.nanoTime() + timeoutInMilliseconds * 1000000L,
-        timeoutInMilliseconds);
+    this.operator = op;
+    this.callback = cb;
+    this.timeoutMs = timeoutInMilliseconds;
+
+    this.enableCounter = enableCounter;
+    this.createNanoTime = System.nanoTime();
```
Comment (Shuo-Jia) [inline src/main/java/com/xiaomi/infra/pegasus/rpc/async/ClientRequestRound.java:42]: `createNanoTime` was deleted in https://github.com/XiaoMi/pegasus-java-client/pull/105, but it is useful in this pr.

File: src/main/java/com/xiaomi/infra/pegasus/metrics/MetricsPool.java [hunk 4]
```diff
@@ -15,11 +15,15 @@
...
 public final class MetricsPool {
+
+  private final String defaultTag;
+
   public MetricsPool(String host, String tags, int reportStepSec) {
     theMetric = new FalconMetric();
     theMetric.endpoint = host;
     theMetric.step = reportStepSec;
     theMetric.tags = tags;
+    defaultTag = tags;
```
Comment (levy5307) [inline src/main/java/com/xiaomi/infra/pegasus/metrics/MetricsPool.java:26]: `defaultTags`

**Pull request merged**